retry.go

```diff
@@ -118,8 +135,8 @@ func (r ExponentialRetry) RetryWithCheck(ctx context.Context, checkFn RetryCheck
 		waitInterval = r.WaitInterval
 	}
 	jitterFraction := .05
-	if r.jitterFraction != 0 {
-		jitterFraction = r.jitterFraction
+	if r.JitterFraction != 0 {
+		jitterFraction = r.JitterFraction
```

Comment:
Github won't let me comment on line #149 but I'm guessing it should be `call duration`

Reply:
if you want to set a time limit, you do it through the context, but a max retries would make sense

```
ctx := context.WithTimeout(parentCtx, 5*time.Second)
err = rc.Retry(ctx, func() error{ ... })
```